01_easy_algorithms/07_recursion_greatest_common_divisor/07_recursion_greatest_common_divisor.rb

```diff
@@ -0,0 +1,33 @@
+def gcd (a, b)
+  
+  if a > b 
+    x = a
+    y = b
+  else 
+    x = b
+    y = a
+  end
+
+  n = x % y
+  
+  while (n != 0) 
+    x = y
+    y = n
+    n = x % y
+  end
+  y
+end
+
+def gcd_recursion(a, b)
+  
+    while a != b
+        if a > b
+            return gcd_recursion(a - b, b);
+        else
+            return gcd_recursion(a, b - a);
+        end
+    end
+    
+    return a;
```

Comment:
You don't need to specify return here. Just a will do. Ruby automatically returns the last object in a method.